src/cli-sdk/src/commands/version.ts

```diff
@@ -0,0 +1,230 @@
+import {
+  inc,
+  parse as parseVersion,
+  versionIncrements,
+} from '@vltpkg/semver'
+import type { IncrementType } from '@vltpkg/semver'
+import { is as isGit, spawn as spawn_, isClean } from '@vltpkg/git'
+import type { GitOptions } from '@vltpkg/git'
+import { error } from '@vltpkg/error-cause'
+import { asError } from '@vltpkg/types'
+import { commandUsage } from '../config/usage.ts'
+import type { CommandFn, CommandUsage } from '../index.ts'
+import type { Views } from '../view.ts'
+import type { ParsedConfig } from '../config/index.ts'
+
+export type VersionOptions = {
+  prereleaseId?: string
+  commit?: boolean
+  tag?: boolean
+  message?: string
+  tagMessage?: string
+}
+
+export type VersionResult = {
+  oldVersion: string
+  newVersion: string
+  dir: string
+  committed?: string[]
+  tag?: string
+}
+
+const isValidVersionIncrement = (
+  value: string,
+): value is IncrementType =>
+  versionIncrements.includes(value as IncrementType)
+
+const version = async (
+  conf: ParsedConfig,
+  increment: string | undefined,
+  cwd: string,
+  {
+    // Hardcode happy path options for now.
+    // TODO: make these config definitions
+    prereleaseId = 'pre',
+    commit = true,
+    tag = true,
+    message = 'v%s',
+    tagMessage = 'v%s',
+  }: VersionOptions = {},
+): Promise<VersionResult> => {
+  const spawn = (args: string[], opts?: GitOptions) =>
+    spawn_(args, { cwd, ...opts })
+
+  if (!increment) {
+    throw error('Version increment argument is required', {
+      code: 'EUSAGE',
+      validOptions: versionIncrements,
+    })
+  }
+
+  const manifest = conf.options.packageJson.read(cwd)
```

Comment:
I just added a very handy method in https://github.com/vltpkg/vltpkg/pull/940 that we can potentially use here instead:
```suggestion
  const manifest = conf.options.packageJson.find()
```

That's going to walk up the directory tree and operate on the first found `package.json` file, which is going to allow this command to also work from nested directories